codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -194,7 +237,8 @@ private void flushBufferedData(ByteBuf out) {
         checksum.update(buffer.array(), buffer.arrayOffset(), currentBlockLength);
         final int check = (int) checksum.getValue();
 
-        out.ensureWritable(compressedBlockSize);
+        final int bufSize = compressor.maxCompressedLength(buffer.readableBytes()) + HEADER_LENGTH;
+        out.ensureWritable(bufSize);
```

Comment:
is it possible the compression will take less than `bufSize`? If so we may fail the compression upfront if the `out` buffer has a `maxCapacity` set which would accommodate the real compression size, but can't accommodate the max `bufSize` size.